### PR TITLE
Feature - Taipy middleware

### PR DESCRIPTION
**Pull request by khaitruong922**
Middleware for handling community and enterprise version.
Refactor to use manager factory.

Refer to this PR as well: https://github.com/Avaiga/taipy-enterprise/pull/57

### CODE REVIEW

**Review by jrobinAV — COMMENTED** [review]
Next time, please put the side changes not related to the feature to implement in another PR (like the `.isort.cfg`, `.pre-commit-config.yaml`, etc...). In particular for PRs like this one with more than a few files changed.
It makes it difficult to review.

File: src/taipy/rest/api/middlewares/_taipy_middleware.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from flask import request
+
+from taipy.core.common._utils import _load_fct
+
+
+def _taipy_middleware(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if _using_enterprise():
+            print("Enterprise is installed")
```
Comment (jrobinAV) [inline src/taipy/rest/api/middlewares/_taipy_middleware.py]: Please replace by a taipylogger.debug.

File: src/taipy/rest/api/middlewares/_taipy_middleware.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            return _enterprise_middleware()(request)(f)(*args, **kwargs)
+        else:
+            print("Enterprise not installed")
```
Comment (jrobinAV) [inline src/taipy/rest/api/middlewares/_taipy_middleware.py]: Please replace by a taipylogger.debug.

File: src/taipy/rest/api/middlewares/_taipy_middleware.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+from functools import wraps
+from importlib import util
+
+from flask import request
+
+from taipy.core.common._utils import _load_fct
+
+
+def _taipy_middleware(f):
```
Comment (jrobinAV) [inline src/taipy/rest/api/middlewares/_taipy_middleware.py]: I am not big fan of the name, but I don't have better name to propose at this point.

File: src/taipy/rest/api/middlewares/_taipy_middleware.py (hunk 1 above)
Comment (khaitruong922) [reply src/taipy/rest/api/middlewares/_taipy_middleware.py]: This is just for debugging, will remove this when finished.

File: src/taipy/rest/api/middlewares/_taipy_middleware.py (hunk 3 above)
Comment (khaitruong922) [reply src/taipy/rest/api/middlewares/_taipy_middleware.py]: How about just `_middleware`?

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**